spec.html

```diff
@@ -30753,19 +30756,23 @@ <h1>String.prototype.replace ( _searchValue_, _replaceValue_ )</h1>
               1. Return ? Call(_replacer_, _searchValue_, &laquo; _O_, _replaceValue_ &raquo;).
           1. Let _string_ be ? ToString(_O_).
           1. Let _searchString_ be ? ToString(_searchValue_).
+          1. Let _searchLen_ be the length of _searchString_.
           1. Let _functionalReplace_ be IsCallable(_replaceValue_).
           1. If _functionalReplace_ is *false*, then
             1. Set _replaceValue_ to ? ToString(_replaceValue_).
-          1. Search _string_ for the first occurrence of _searchString_ and let _pos_ be the index within _string_ of the first code unit of the matched substring and let _matched_ be _searchString_. If no occurrences of _searchString_ were found, return _string_.
+          1. Let _pos_ be ! StringIndexOf(_string_, _searchString_, 0).
+          1. If _pos_ is -1, return _string_.
           1. If _functionalReplace_ is *true*, then
-            1. Let _replValue_ be ? Call(_replaceValue_, *undefined*, &laquo; _matched_, _pos_, _string_ &raquo;).
+            1. Let _replValue_ be ? Call(_replaceValue_, *undefined*, &laquo; _searchString_, _pos_, _string_ &raquo;).
             1. Let _replStr_ be ? ToString(_replValue_).
           1. Else,
             1. Let _captures_ be a new empty List.
-            1. Let _replStr_ be ! GetSubstitution(_matched_, _string_, _pos_, _captures_, *undefined*, _replaceValue_).
-          1. Let _tailPos_ be _pos_ + the number of code units in _matched_.
-          1. Let _newString_ be the string-concatenation of the first _pos_ code units of _string_, _replStr_, and the trailing substring of _string_ starting at index _tailPos_. If _pos_ is 0, the first element of the concatenation will be the empty String.
-          1. Return _newString_.
+            1. Let _replStr_ be ! GetSubstitution(_searchString_, _string_, _pos_, _captures_, *undefined*, _replaceValue_).
+          1. Let _tailPos_ be _pos_ + _searchLen_.
+          1. Return the string-concatenation of:
+            * the substring of _string_ from 0 to _pos_
```

Comment:
I don't think we need to create a list here for this. There's not a huge number of items (3) and they're not that long. Compare with this step below:

>  1. Set _accumulatedResult_ to the string-concatenation of the current value of _accumulatedResult_, the substring of _S_ from _nextSourcePosition_ to _position_, and _replacement_.